lib/utils/matrix_sdk_extensions/matrix_file_extension.dart

```diff
@@ -61,6 +38,42 @@ extension MatrixFileExtension on MatrixFile {
     return;
   }
 
+  void downloadImageInWeb(BuildContext context) {
+    Logs().d("MatrixFileExtension()::downloadImageInWeb()::download on Web");
+
+    html.AnchorElement(
+      href: html.Url.createObjectUrlFromBlob(
+        html.Blob(
+          [bytes],
+          mimeType,
+        ),
+      ),
+    )
+      ..download = name
+      ..click();
+
+    Fluttertoast.showToast(
+      msg: L10n.of(context)!.downloadImageInWeb,
+    );
+  }
+
+  void downloadImageInMobile(BuildContext context) async {
+    Logs().d(
+      "MatrixFileExtension()::downloadImageInMobile()::download on Mobile",
+    );
+
+    final result = await ImageGallerySaver.saveImage(
+      bytes ?? Uint8List(0),
+      name: name,
+    );
+
+    Fluttertoast.showToast(
+      msg: result?['isSuccess']
```

Comment:
If `result is null`, what is happen?

Reply:
if result is null it will show the error toast

Reply:
still not clear for me, what will handle the case of `result is null`?

Reply:
the null safety will handle this case 

![image](https://github.com/linagora/twake-on-matrix/assets/48354990/9e19467f-46a2-481c-94bc-c3b4df40fc52)

Reply:
it's good for me